### DISCUSSION TITLE
Best practice of rear cable on patch panel

### DISCUSSION THREAD

**Discussion by LLFT** [netbox-community/netbox#15864] [General]
Hello,

Question of usage.

 When you create two copper or fiber patch panel, do you create one rear cable that connects all of the rear ports to the other rear ports of the other patch panel? Or do you create one cable for each of the rear ports?

Thx !

**Comment by candlerb** [comment]
If you have N front ports linked to N rear ports, then I would suggest you make N separate rearport-to-rearport cables. This is so that you can trace through them properly. There is a helper script [here](https://github.com/netbox-community/customizations/blob/master/scripts/multi_connect.py).

If instead you decided to create 1 cable with N terminations at each end, then the terminations are not associated with each other in any particular order (as far as I know), which I suspect means cable tracing wouldn't work (although you can try it). Indeed, there's no requirement even for a cable to have an equal number of terminations at each end.

If instead you have N front ports linked to 1 rear port with N positions, then you just make a single cable between this rear port and the remote one, and cable tracing *does* work correctly.

**Reply by LLFT** [reply]
Thanks for the script

**Comment by Chiniquy** [comment]
Both. 
Fiber patch panels typically have 1 physical fiber cable, so having 1 rear port map to all the front ports makes sense. 
Copper patch panels have many physical cables in the back, one for each front port, so having a 1-to-1 mapping between front and rear ports makes sense.

**Reply by LLFT** [reply]
I agreed with you until I tested

**Comment by LLFT** [comment]
Thank you for your feedback

I didn't understand the point in the rear ports of the "Position" parameter. I thought it was to define the OM4 fibers which feed each port section, but that doesn't seem to be that.

Otherwise, I did a test to see the difference between the two uses. A cable for everyone and everyone a cable (LOL)

It’s totally possible! Where it no longer works is when we have to go through a rear cable for everyone. It is impossible to associate a front interface. This gives me an error like Class AssertionError Plugins: None installed.

**Reply by candlerb** [reply]
> I didn't understand the point in the rear ports of the "Position" parameter. I thought it was to define the OM4 fibers which feed each port section, but that doesn't seem to be that.

It kind-of is, but it's logical connections rather than actual fibre strands (e.g. one "position" could actually be a pair of fibres in real life)

Let's say you create a device type with 1 rear port which has 12 positions. You can then create 12 front ports linked to the same rear port but positions :1 to :12 (these can created in one step, entering a name pattern like `F[1-12]` and selecting all 12 positions)

Then you can connect two of these patch-panels back-to-back, with a single cable connecting patch panel A's rear port to patch panel B's rear port. This might represent a 24-strand MPO cable in real life.

>  It is impossible to associate a front interface. This gives me an error like Class AssertionError Plugins: None installed.

Sorry, I have no idea what you did to make that error occur, but plugins should not come into the picture at all.  You'd need to give full details of your setup (e.g. what version of Netbox and Python) and the exact sequence of steps to reproduce the error.
